engine/tests/classes/Elgg/IntegrationTestCase.php

```diff
@@ -143,18 +118,26 @@ public static function tearDownAfterClass() {
 	final protected function tearDown() {
 		$this->down();
 
-		if ($this instanceof LegacyIntegrationTestCase) {
-			_elgg_services()->session->removeLoggedInUser();
-		}
-
+		$app = Application::getInstance();
 
-		$this->assertEquals($this->_testing_hooks, _elgg_services()->hooks->getAllHandlers(), "
-			The test has failed to clean up hook registrations after itself.
-		");
+		foreach ($this->_testing_hooks as $type => $hooks) {
+			foreach ($hooks as $hook => $handlers) {
+				foreach ($handlers as $handler) {
+					$this->assertTrue($app->_services->hooks->hasHandler($type, $hook, $handler));
```

Comment:
I tried that and there are some hooks that use closures with injected session and various timestamps, so it doesn't work